tests/subsys/usb_c/mocks/mock_tester.c

```diff
@@ -0,0 +1,393 @@
+/*
+ * Copyright 2023 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <string.h>
+#include <stddef.h>
+#include <zephyr/drivers/usb_c/usbc_pd.h>
+
+#include "mock_tester.h"
+#include "mock_tcpc.h"
+
+/**
+ * @breif Tester data
+ */
+struct mock_tester_t {
+	/* pointer to mock tcpc */
+	struct mock_tcpc_data *tcpc;
+	/* CC1 voltage state */
+	enum tc_cc_voltage_state cc1;
+	/* CC2 voltage state */
+	enum tc_cc_voltage_state cc2;
+	/* CC Pull set by UUT */
+	enum tc_cc_pull uut_cc_pull;
+	/* CC Polarity set by UUT */
+	enum tc_cc_polarity uut_polarity;
+
+	/* VBUS set by UUT or Tester */
+	int vbus;
+	/* PD Message to send to the UUT */
+	struct pd_msg tx_msg;
+	/* PD Message received from the UUT */
+	struct pd_msg rx_msg;
+
+	/* PD message ID */
+	int msg_id;
+	/* PD revision */
+	enum pd_rev_type rev;
+	/* Tester's Power Role */
+	enum tc_power_role power_role;
+	/* Tester's Data Role */
+	enum tc_data_role data_role;
+};
+
+/* Test data object */
+static struct mock_tester_t tester;
+
+/**
+ * @brief Set the PD message ID
+ *
+ * @param id The id to set
+ */
+void tester_msgid_set(int id)
+{
+	tester.tx_msg.header.message_id = id & 7;
+}
+
+/**
+ * @brief Increment the PD message ID
+ */
+void tester_msgid_inc(void)
+{
+	tester.tx_msg.header.message_id++;
+	tester.tx_msg.header.message_id &= 7;
+}
+
+/**
+ * @brief Get the PD message ID
```

Comment:
These comments appear to duplicate the comments in the header file. I think you can remove them and just have comments for the public API in the header.